internal/transport/http2_server.go

```diff
@@ -266,6 +266,13 @@ func newHTTP2Server(conn net.Conn, config *ServerConfig) (_ ServerTransport, err
 	// Check the validity of client preface.
 	preface := make([]byte, len(clientPreface))
 	if _, err := io.ReadFull(t.conn, preface); err != nil {
+		// In deployments where a gRPC server runs behind a cloud load balancer
+		// which performs regular TCP level health checks, the connection is
+		// closed immediately by the latter. Skipping the error here will help
+		// reduce log clutter.
+		if err == io.EOF {
+			return nil, nil
```

Comment:
Let's make sure this is documented above and in `NewServerTransport`.  Currently both indicate an error will be returned, or a valid transport.

To make these easier to keep in sync, I'd be happy if we finally delete the current `NewServerTransport` and rename this function to it.  Or use another name (e.g. `NewHTTP2Server[Transport]`) if desired.

Reply:
Done. Ended up retaining the name `NewServerTransport` to match with the existing `NewClientTransport`.